modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageTest.java

```diff
@@ -105,7 +121,94 @@ void testReadAfterRestart() throws Exception {
         tearDown();
 
         setUp();
+    }
 
-        assertRowMatches(binaryRow, read(rowId, HybridTimestamp.MAX_VALUE));
+    @Test
+    void groupConfigIsPersisted() throws Exception {
+        RaftGroupConfiguration originalConfig = new RaftGroupConfiguration(
+                List.of("peer1", "peer2"),
+                List.of("old-peer1", "old-peer2"),
+                List.of("learner1", "learner2"),
+                List.of("old-learner1", "old-learner2")
+        );
+
+        storage.runConsistently(() -> {
+            storage.committedGroupConfiguration(originalConfig);
+
+            return null;
+        });
+
+        restartStorage();
+
+        RaftGroupConfiguration readConfig = storage.committedGroupConfiguration();
+
+        assertThat(readConfig, is(equalTo(originalConfig)));
+    }
+
+    @Test
+    void groupConfigWhichDoesNotFitInOnePageIsPersisted() throws Exception {
+        List<String> oneMbOfPeers = IntStream.range(0, 100_000)
+                .mapToObj(n -> String.format("peer%06d", n))
+                .collect(toList());
+
+        RaftGroupConfiguration originalConfig = new RaftGroupConfiguration(
+                oneMbOfPeers,
+                List.of("old-peer1", "old-peer2"),
+                List.of("learner1", "learner2"),
+                List.of("old-learner1", "old-learner2")
+        );
+
+        storage.runConsistently(() -> {
+            storage.committedGroupConfiguration(originalConfig);
+
+            return null;
+        });
+
+        restartStorage();
+
+        RaftGroupConfiguration readConfig = storage.committedGroupConfiguration();
+
+        assertThat(readConfig, is(equalTo(originalConfig)));
+    }
+
+    @Test
+    void groupConfigPagesAreRecycled() throws Exception {
+        List<String> oneMbOfPeers = IntStream.range(0, 100_000)
```

Comment:
I think it's worth making sure the size of these `Strings` is greater than at least 2 pages.